api/src/main/java/org/pmiops/workbench/workspaces/WorkspacesController.java

```diff
@@ -558,6 +521,33 @@ private FirecloudWorkspaceId generateFirecloudWorkspaceId(String namespace, Stri
             .workspace(workspaceMapper.toApiWorkspace(savedWorkspace, toFcWorkspace)));
   }
 
+  // A retry period is needed because the permission to copy files into the cloned workspace is not granted transactionally
+  private void copyBlobWithRetries(String bucketName, Blob b, int retryPeriodSeconds) {
+    final int SLEEP_INTERVAL_SECONDS = 5;
+
+    try {
+      cloudStorageService.copyBlob(b.getBlobId(), BlobId.of(bucketName, b.getName()));
+    } catch (StorageException e) {
+      if (retryPeriodSeconds <= 0) {
```

Comment:
I think this name is misleading. I'd thought a retry period would be the interval of retrying. As used, it's really a total pool of time, and the interval is `retryPeriodSeconds / SLEEP_INTERVAL_SECONDS`. Likewise, `SLEEP_INTERVAL_SECONDS` is confusing.

I recommend having a separate parameter for `numRetries` and then just a fixed duration for the interval.

Even better would be to use a dedicated `Retryer` class like [this one](https://www.programcreek.com/java-api-examples/?api=com.github.rholder.retry.Retryer).

Reply:
Added. Looks like we already use Spring's `Retryable` in a few places but I decided to add the new library since there was some discussion of being less tied to Spring.

Reply:
Nice. I _believe_ this was the one we used at another company.